webapp/src/containers/LinkModal/index.tsx

```diff
@@ -1,138 +1,126 @@
-        linkProjectError: '',
+
+    const [errorState, setErrorState] = useState<LinkPayload>({
+        organization: '',
+        project: '',
     });
-    const [linkPayload, setLinkPayload] = useState<LinkPayload | null>();
+
+    // Hooks
     const usePlugin = usePluginApi();
-    const {visibility, organization, project} = usePlugin.state['plugins-mattermost-plugin-azure-devops'].openLinkModalReducer;
-    const [loading, setLoading] = useState(false);
-    const [APIError, setAPIError] = useState('');
     const dispatch = useDispatch();
 
-    useEffect(() => {
-        if (organization && project) {
-            setState({
-                linkOrganization: organization,
-                linkProject: project,
-            });
-        }
-    }, [visibility]);
-
     // Function to hide the modal and reset all the states.
-    const onHide = useCallback(() => {
-        setState({
-            linkOrganization: '',
-            linkProject: '',
+    const resetModalState = () => {
+        setProjectDetails({
+            organization: '',
+            project: '',
         });
-        setError({
-            linkOrganizationError: '',
-            linkProjectError: '',
+        setErrorState({
+            organization: '',
+            project: '',
         });
-        setLinkPayload(null);
-        setLoading(false);
-        setAPIError('');
         dispatch(hideLinkModal());
-    }, []);
+    };
+
+    // Set organization name
+    const onOrganizationChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        setErrorState({...errorState, organization: ''});
+        setProjectDetails({...projectDetails, organization: (e.target as HTMLInputElement).value});
+    };
 
-    const onOrganizationChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
-        setError({...error, linkOrganizationError: ''});
-        setState({...state, linkOrganization: (e.target as HTMLInputElement).value});
-    }, [state, error]);
+    // Set project name
+    const onProjectChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        setErrorState({...errorState, project: ''});
+        setProjectDetails({...projectDetails, project: (e.target as HTMLInputElement).value});
+    };
 
-    const onProjectChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
-        setError({...error, linkProjectError: ''});
-        setState({...state, linkProject: (e.target as HTMLInputElement).value});
-    }, [state, error]);
+    // Handles on confirming link project
+    const onConfirm = () => {
+        const errorStateChanges: LinkPayload = {
+            organization: '',
+            project: '',
+        };
 
-    const onConfirm = useCallback(() => {
-        if (state.linkOrganization === '') {
-            setError((value) => ({...value, linkOrganizationError: 'Organization is required'}));
+        if (projectDetails.organization === '') {
+            errorStateChanges.organization = 'Organization is required';
```

Comment:
i think we can create constants for these errors

Reply:
we need to create a common validation point for all forms in web app as `required, invalid, etc` are needed in every forms so, instead of creating a constant for each such form field we can create a central validation service

created ticket we can target this as enhancement https://brightscout.atlassian.net/jira/software/c/projects/MI/boards/13?selectedIssue=MI-2050